File: sdk/python/foodblock/tombstone.py
"""Tombstone block creation for content erasure (Section 5.4)."""
import uuid
from datetime import datetime, timezone
from .block import create

def tombstone(target_hash: str, requested_by: str, reason: str = "erasure_request") -> dict:
    """Create a tombstone block that marks a target for content erasure."""
    if not target_hash:
        raise ValueError("FoodBlock: target_hash is required")
    if not requested_by:
        raise ValueError("FoodBlock: requested_by is required")
    return create("observe.tombstone", {
        "instance_id": str(uuid.uuid4()),
        "reason": reason,
        "requested_by": requested_by,
        "requested_at": datetime.now(timezone.utc).isoformat(),
    }, {
        "target": target_hash,
        "updates": target_hash,
    })

File: sdk/javascript/src/tombstone.js
const { create } = require('./block')
const crypto = require('crypto')

/**
 * Create a tombstone block that marks a target block for content erasure.
 * See Section 5.4 of the whitepaper.
 *
 * @param {string} targetHash - Hash of the block to erase
 * @param {string} requestedBy - Hash of the actor requesting erasure
 * @param {object} [opts] - { reason: string }
 * @returns {object} - The tombstone FoodBlock
 */
function tombstone(targetHash, requestedBy, opts = {}) {
  if (!targetHash || typeof targetHash !== 'string') {
    throw new Error('FoodBlock: targetHash is required')
  }
  if (!requestedBy || typeof requestedBy !== 'string') {
    throw new Error('FoodBlock: requestedBy is required')
  }

  return create('observe.tombstone', {
    instance_id: crypto.randomUUID(),
    reason: opts.reason || 'erasure_request',
    requested_by: requestedBy,
    requested_at: new Date().toISOString()
  }, {
    target: targetHash,
    updates: targetHash
  })
}

module.exports = { tombstone }
